### ISSUE TITLE
JVM intermittently crashes partway through integration tests

### ISSUE THREAD

**Issue by gracegrimwood** [kroxylicious/kroxylicious#508]
**Describe the bug**
Sometimes when running the failsafe plugin stage on the integrationtests module via maven, the JVM will crash without warning, i.e.
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:3.1.2:verify (default) on project integrationtests: 
[ERROR] 
[ERROR] Please refer to /Users/ggrimwoo/workspace/temp/kroxylicious/integrationtests/target/failsafe-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd '/Users/ggrimwoo/workspace/temp/kroxylicious/integrationtests' && '/Users/ggrimwoo/.sdkman/candidates/java/20-tem/bin/java' '-jar' '/Users/ggrimwoo/workspace/temp/kroxylicious/integrationtests/target/surefire/surefirebooter-20230726152737515_3.jar' '/Users/ggrimwoo/workspace/temp/kroxylicious/integrationtests/target/surefire' '2023-07-26T15-27-37_485-jvmRun1' 'surefire-20230726152737515_1tmp' 'surefire_0-20230726152737515_2tmp'
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] Crashed tests:
[ERROR] io.kroxylicious.proxy.ExpositionIT
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:643)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:285)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:250)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1203)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1055)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:871)
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:342)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:330)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:175)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:76)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:163)
[ERROR] 	at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:160)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:910)
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:283)
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:206)
[ERROR] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:283)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:226)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:407)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:348)
```

**To Reproduce**
1. Run `mvn verify -pl :integrationtests` or any other maven command that runs the integration tests
2. Sometimes the JVM will crash with a message like above.

**Expected behavior**
The JVM should not crash during integration tests (or ever, ideally).

**YAML files and logs**

[Crash on branch multitenant-util-methods-450](https://gist.github.com/gracegrimwood/5bd9d4978cd3857948cd0942b5b68112) (for PR #490)
[Crash on main](https://gist.github.com/gracegrimwood/5bd9d4978cd3857948cd0942b5b68112)
[Crash on branch using Security Manager](https://gist.github.com/gracegrimwood/bdfd5ebc29395dbfdb655b39eada8b1d) to intercept calls to `System.exit()`

**Additional context**
Testing with @robobario and @SamBarker showed that this is an actual JVM crash, not something silently exiting. This issue can be reproduced on the most recent commit on main.
The Gists include lines like `=============== RUN 1 STARTING ===============`, this is because the tests were run repeatedly until failure by a script that Sam wrote. The version of that script that I'm using is below:
```bash
#!/usr/bin/env bash
#
# Copyright Kroxylicious Authors.
#
# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
#

targetTest=${1:-ExpositionIT}

logFile=/tmp/mvn-${targetTest}-$(date +"%d-%m-%Y_%H:%M").log

mvn clean install -DskipTests=true

for COUNT in {1..10} ; do
    echo "=============== RUN ${COUNT} STARTING =============== " | tee -a "${logFile}"
    mvn verify -Dit.test=${targetTest} -pl :integrationtests >> "${logFile}"
    EC=$?
    echo "=============== RUN ${COUNT} FINISHED =============== " | tee -a "${logFile}"
    if [[ ${EC} -ne 0 ]]; then
      exit $EC
    fi
done
```
Originally we were targeting `ExpositionIT` as that's where we'd seen this happen, but I've since had it occur on other integration tests.

**Comment by gracegrimwood** [comment]
Worth noting this bug is partially masked by other general instability in the integration tests; it seems they all randomly fail _without crashing_ with a wide variety of errors. Not sure if we should roll those into this issue or create a separate one for those "less serious" problems.

**Comment by gracegrimwood** [comment]
Tried again with Kafka logging turned up, seems Kafka tries (and succeeds?) to do a clean shutdown right before the JVM crashes. Logs in [this gist](https://gist.github.com/gracegrimwood/972b2ae7a8bd6ab4ed1fecdcbf890c96).

**Comment by gracegrimwood** [comment]
This looks to originate in the kroxylicious-junit-extension. Continuing the search there.

**Comment by SamBarker** [comment]
We think this is triggered by https://github.com/kroxylicious/kroxylicious-junit5-extension/issues/162 which has been fixed so this will be resolved by the next release of the JUnit extension.

**Comment by gracegrimwood** [comment]
This was fixed in #588.
